tests/interfaces/records-read.spec.ts

```diff
@@ -22,6 +22,51 @@ describe('RecordsRead', () => {
 
       expect(recordsRead.message.descriptor.messageTimestamp).to.equal(currentTime);
     });
+
+    it('should reject if `recordId`, `protocol` and `protocolPath` are all missing', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+      const readPromise = RecordsRead.create({
+        authorizationSignatureInput: Jws.createSignatureInput(alice),
+      });
+
+      await expect(readPromise).to.be.rejectedWith(DwnErrorCode.RecordsReadMissingCreateProperties);
+    });
+
+    it('should not reject if only `recordId` is passed', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+      const readPromiseSuccess = RecordsRead.create({
+        recordId                    : 'some-id',
+        authorizationSignatureInput : Jws.createSignatureInput(alice),
+      });
+
+      await expect(readPromiseSuccess).to.not.be.rejected;
+    });
+
+    it('should reject if only one of `protocol` or `protocolPath` are set', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+      // with only protocolPath
+      const protocolPathOnlyP = RecordsRead.create({
```

Comment:
What is the `P` at the end of `protocolPathOnlyP` and `protocolOnlyP`?

Reply:
The `P` stands for `Promise`, I usually put that as a naming convention when I haven't awaited the promise.

I can avoid that if you think it adds confusion (or use the full word).

Reply:
I'd go for full word.